### PR TITLE
React to directive error squiggly fix 

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Related PR - aspnet/Razor#392
@NTaylorMullen @dougbu

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorCodeParser.cs [hunk 1]
```diff
@@ -50,17 +50,19 @@ protected virtual void ModelDirective()
         {
             // Verify we're on the right keyword and accept
             AssertDirective(ModelKeyword);
+            var startModelLocation = CurrentLocation;
             AcceptAndMoveNext();
+            var keywordLength = Span.GetContent().Value.Length;
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorCodeParser.cs]: `ModelKeyword.Length`

Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorCodeParser.cs]: And move down just before it's used.  No reason to calculate at all unless adding an error.

Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorCodeParser.cs]: :+1:

**Comment by NTaylorMullen** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorCodeParser.cs [hunk 2]
```diff
@@ -95,8 +100,10 @@ protected virtual void InjectDirective()
             var hasTypeError = !At(CSharpSymbolType.Identifier);
             if (hasTypeError)
             {
-                Context.OnError(CurrentLocation,
-                                Resources.FormatMvcRazorCodeParser_KeywordMustBeFollowedByTypeName(InjectKeyword));
+                Context.OnError(
+                    startLocation,
+                    Resources.FormatMvcRazorCodeParser_KeywordMustBeFollowedByTypeName(InjectKeyword),
+                    keywordwithSingleWhitespaceLength + remainingWhitespaceLength);
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorCodeParser.cs]: Be consistent.  Not including any whitespace in the `model` error.

File: src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorCodeParser.cs [hunk 3]
```diff
@@ -105,14 +112,14 @@ protected virtual void InjectDirective()
...
             AcceptWhile(IsSpacingToken(includeNewLines: false, includeComments: true));
 
             if (!hasTypeError && (EndOfFile || At(CSharpSymbolType.NewLine)))
             {
                 // Add an error for the property name only if we successfully read the type name
-                Context.OnError(propertyStartLocation,
-                                Resources.FormatMvcRazorCodeParser_InjectDirectivePropertyNameRequired(InjectKeyword));
+                Context.OnError(startLocation,
+                                Resources.FormatMvcRazorCodeParser_InjectDirectivePropertyNameRequired(InjectKeyword),
+                                keywordwithSingleWhitespaceLength + remainingWhitespaceLength);
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorCodeParser.cs]: This is going to look strange.  Suggest the error include all of the directive that was present -- the type name as well as the keyword?

**Comment by dougbu** [comment]
General thought: In an erroneous line such as `@model  `, errors will highlight `model` with this change.  Would including the entire line or at least the leading `@` make more sense?  (Highlighting either `@model  ` or `@model` here.)

Could / should we do that for all errors within directives?

**Comment by dougbu** [comment]
:watch:

**Comment by NTaylorMullen** [comment]
Whatever we do for `@model` we should do for `@inherits`

**Comment by ajaybhargavb** [comment]
@dougbu, I don't think highlighting the leading `@` is a good idea because it won't be consistent with other errors. Eg: When you type `@abc` without creating the variable `abc` it is going to highlight `abc` and not `@abc`.

**Comment by NTaylorMullen** [comment]
oo I agree with @ajaybhargavb here, we don't highlight the `@` for any other directive and never have

File: src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorCodeParser.cs (hunk 2 above)
Comment (ajaybhargavb) [inline src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorCodeParser.cs]: :+1: I am removing whitespace from this error. Will also change `@inherits` to not include the trailing whitespace.

**Comment by ajaybhargavb** [comment]
Updated.

**Comment by dougbu** [comment]
:shipit:

**Pull request merged**